### PR TITLE
Implement Restore Wallet view

### LINKED ISSUE #423: Implement Restore wallet from recovery phrase
Basically reuses the Verify seed design: https://projects.invisionapp.com/d/main/default/#/console/13241316/281903495/preview

The view would ask the user to input the seed words in random order (to mitigate key loggers).

### PR DESCRIPTION
**Pull request by valentinewallace**
Closes #423 
Let me know if you think the abstraction isn't good

### CODE REVIEW

File: src/view/restore-wallet.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+const RestoreWalletView = ({ store, nav, wallet }) => (
+  <SeedInputView
+    title="Let's restore your wallet"
+    copy={store.seedVerifyCopy}
+    navBack={nav.goSeed}
+    seedInputs={store.wallet.seedVerify}
+    indexes={store.seedVerifyIndexes}
+    setSeedVerify={wallet.setSeedVerify}
+    checkSeed={wallet.checkSeed}
+  />
```
Comment (tanx) [inline src/view/restore-wallet.js]: A view inside of another view 😃 We should probably rename this to a just `SeedInput`.

recursion

File: src/component/seed-input.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+SeedInputView.propTypes = {
+  title: PropTypes.string.isRequired,
+  copy: PropTypes.string,
+  navBack: PropTypes.func.isRequired,
+  seedInputs: PropTypes.array.isRequired,
+  indexes: PropTypes.array.isRequired,
+  setSeedVerify: PropTypes.func.isRequired,
+  checkSeed: PropTypes.func.isRequired,
+};
```
Comment (tanx) [inline src/component/seed-input.js]: Getting prop warnings in the console in storybook.

File: src/view/restore-wallet.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import PropTypes from 'prop-types';
+import SeedInputView from '../component/seed-input';
+
+//
+// Restore Wallet View
+//
+
+const RestoreWalletView = ({ store, nav, wallet }) => (
+  <SeedInputView
+    title="Let's restore your wallet"
```
Comment (tanx) [inline src/view/restore-wallet.js]: Shorten to `Restore your wallet`

File: test/unit/action/nav.spec.js [hunk 4]
```diff
@@ -39,6 +39,13 @@ describe('Action Nav Unit Tests', () => {
     });
   });
 
+  // describe('goRestoreWallet()', () => {
+  //   it('should set correct route', () => {
+  //     nav.goRestoreWallet();
+  //     expect(store.route, 'to equal', 'RestoreWallet');
+  //   });
+  // });
+
```
Comment (tanx) [inline test/unit/action/nav.spec.js]: You can `describe.skip(...)` tests.

File: src/action/nav.js [hunk 5]
```diff
@@ -15,6 +15,10 @@ class NavAction {
     this._store.route = 'SeedVerify';
   }
 
+  goRestoreWallet() {
+    // this._store.route = 'Restore Wallet';
```
Comment (tanx) [inline src/action/nav.js]: 'Restore Wallet' without the space

File: src/component/seed-input.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+import { Header } from '../component/header';
+import { H1Text, Text } from '../component/text';
+import { Button, BackButton, GlasButton } from '../component/button';
+import { InputField } from '../component/field';
+import Card from '../component/card';
+import { FormSubText } from '../component/form';
+import { color, font } from '../component/style';
+
+//
+// Generic Seed Input View
```
Comment (tanx) [inline src/component/seed-input.js]: No `View`

File: src/component/seed-input.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+            key={i}
+            autoFocus={i === 0}
+            onSubmitEditing={() => checkSeed()}
+          />
+        ))}
+      </Card>
+      <GlasButton onPress={() => checkSeed()}>Next</GlasButton>
+    </MainContent>
+  </Background>
+);
```
Comment (tanx) [inline src/component/seed-input.js]: Having taken another look at this component, I'm not sure it's a good abstraction to bundle all the contents of a view in a component. I'd suggest to export/reuse just the `SeedEntry` and rename this file `seed-entry.js`.

File: src/view/seed-verify.js [hunk 8]
```diff
@@ -41,7 +40,7 @@ const SeedVerifyView = ({ store, nav, wallet }) => (
     </Header>
     <MainContent style={styles.content}>
       <View>
-        <H1Text style={styles.title}>{"Let's double check"}</H1Text>
+        <H1Text style={styles.title}>Restore your wallet</H1Text>
```
Comment (tanx) [inline src/view/seed-verify.js]: Are you sure this is what we want in the seed-verify screen? Seems like the copy the way it was is ok.

File: src/view/main.js [hunk 9]
```diff
@@ -60,6 +61,9 @@ class MainView extends Component {
         {route === 'SeedVerify' && (
           <SeedVerify store={store} nav={nav} wallet={wallet} />
         )}
+        {route === 'RestoreWallet' && (
+          <RestoreWallet store={store} nav={nav} wallet={wallet} />
+        )}
```
Comment (tanx) [inline src/view/main.js]: I'd suggest we wait with inserting the new screen into main.js until we have the logic wired up.

File: src/view/seed-verify.js (hunk 8 above)
Comment (valentinewallace) [reply src/view/seed-verify.js]: My mistake, fixed

File: src/view/main.js (hunk 9 above)
Comment (valentinewallace) [reply src/view/main.js]: fixed :)

**Pull request merged**